src/Temporalio/Worker/WorkflowInstance.cs

```diff
@@ -237,6 +246,9 @@ public WorkflowInstance(WorkflowInstanceDetails details)
         /// <inheritdoc />
         public IReadOnlyDictionary<string, IRawValue> Memo => memo.Value;
 
+        /// <inheritdoc />
+        public IMetricMeter MetricMeter => metricMeter.Value;
```

Comment:
Do we really want to expose a way to record metrics directly from the workflow? I can certainly see how it's useful but I definitely don't love adding things that do side effects in the middle of workflow code.

Reply:
Yes, but it is replay safe just like loggers and tracing. Our other SDKs have this and is a primary parity point we want to achieve with this project IMO.

Reply:
Makes sense